### DISCUSSION TITLE
Map Enter and other special keys

### DISCUSSION THREAD

**Discussion by BTrey** [sxyazi/yazi#1647] [1. Q&A] [answered]
I'd like to use the smart-enter tip but apply it to my Enter key. I followed the example and used 'on = "\<CR\>"' instead of 'on = "l"' but that doesn't seem to work. I've checked the keymap.toml page from the documentation and can't find anything that documents how to refer to non-character keys in a keymap.

**Comment by xfzv** [comment, marked as answer]
Shouldn't it be `<Enter>` instead of `<CR>`?

**Reply by BTrey** [reply]
That seems to work. I think I tried it with a lowercase 'e' and it didn't work. It would be really nice to have a list of the <Names> of all the special keys in the documentation.

**Reply by xfzv** [reply]
> That seems to work. I think I tried it with a lowercase 'e' and it didn't work.

`"e"` should work just fine, are you sure it's not already used by another binding?

> It would be really nice to have a list of the of all the special keys in the documentation.

[The default `keymap.toml`](https://github.com/sxyazi/yazi/blob/49639aa34ca69fc72008dbae6a2ad033d888805f/yazi-config/preset/keymap.toml) can help to a certain extent considering some bindings already use some "special" keys such as `PageUp`/`PageDown` or `Home` / `End`. At least, you would have noticed it was `<Enter>` instead of `<CR>`: https://github.com/search?q=repo%3Asxyazi%2Fyazi%20%3CEnter%3E%20path%3Akeymap.toml&type=code

Opening the help is even better because it shows bindings for your currently running Yazi config:

https://github.com/sxyazi/yazi/blob/49639aa34ca69fc72008dbae6a2ad033d888805f/yazi-config/preset/keymap.toml#L149-L151
